python/sglang/srt/managers/scheduler.py

```diff
@@ -909,13 +937,17 @@ def update_running_batch(self):
     def run_batch(self, batch: ScheduleBatch):
         """Run a batch."""
         self.forward_ct += 1
-
         if self.is_generation:
-            model_worker_batch = batch.get_model_worker_batch()
             if batch.forward_mode.is_decode() or batch.extend_num_tokens != 0:
-                logits_output, next_token_ids = self.tp_worker.forward_batch_generation(
-                    model_worker_batch
-                )
+                if self.server_args.speculative_algorithm.is_not_none():
+                    logits_output, next_token_ids, model_worker_batch = (
+                        self.draft_worker.forward_batch_speculative_generate(batch)
```

Comment:
`model_worker_batch` is the input, not output. If you want to return some output, create a new data structure for it. Use correct name to make it clear whether it is an input or an output of a function.